modules/standard/IO.chpl

```diff
@@ -2134,50 +2115,158 @@ record channel {
+immediate effect. Use :proc:`channel.flush` to control this buffering.
+
+The :record:`channel` type is implementation-defined.
+A value of the :record:`channel` type refers to the state that is used
+to implement the channel operations.
+
+When a :record:`channel` formal argument has default intent, the actual is passed
+by ``const ref`` to the formal upon a function call, and the formal
+cannot be assigned within the function.
+
+The default value of the :record:`channel` type is not associated
+with any file, and so cannot be used to perform I/O.
+
+The :record:`channel` type is generic.
+
+The :record:`channel` type supports 3 fields:
+
+ * ``param writing: bool``:
+     writing is a boolean indicating whether the channels of this type
+     support writing (when `true`) or reading (when `false`).
+
+ *  ``param kind:iokind``:
+     kind is an enum :type:`iokind` that allows narrowing
+     this channel's I/O style for more efficient binary I/O.
+
+  * ``param locking:bool``:
+     locking is a boolean indicating whether it is safe to use this
+     channel concurrently (when `true`).
+
+.. note::
+
+  The ``channel`` type is deprecated. Please use ``fileReader`` or
+  ``fileWriter`` instead.
+ */
+deprecated "channel type is deprecated - use fileReader or fileWriter instead"
+type channel = _channel;
+
 pragma "no doc"
-operator channel.=(ref lhs:channel, rhs:channel) {
+operator _channel.=(ref lhs:_channel, rhs:_channel) {
   if lhs.writing==true && rhs.writing==false {
     compilerError("cannot assign writing channel to reading channel");
   } else if lhs.writing==false && rhs.writing == true {
     compilerError("cannot assign reading channel to writing channel");
   }
 
   // retain -- release
-  on rhs.home {
+  on rhs._home {
     qio_channel_retain(rhs._channel_internal);
   }
 
-  on lhs.home {
+  on lhs._home {
     qio_channel_release(lhs._channel_internal);
   }
 
-  lhs.home = rhs.home;
+  lhs._home = rhs._home;
   lhs._channel_internal = rhs._channel_internal;
 }
 
 pragma "no doc"
-proc channel.init(param writing:bool, param kind:iokind, param locking:bool) {
+proc _channel.init(param writing:bool, param kind:iokind, param locking:bool) {
   this.writing = writing;
   this.kind = kind;
   this.locking = locking;
 }
 
 pragma "no doc"
-proc channel.init(x: channel) {
-  compilerWarning("new channel(otherChannel) is deprecated");
+proc _channel.init(x: _channel) {
+  compilerWarning("new _channel(otherChannel) is deprecated");
   this.writing = x.writing;
   this.kind = x.kind;
   this.locking = x.locking;
-  this.home = x.home;
+  this._home = x._home;
   this._channel_internal = x._channel_internal;
-  _readWriteThisFromLocale = x._readWriteThisFromLocale;
+  this._readWriteThisFromLocale = x._readWriteThisFromLocale;
   this.complete();
-  on x.home {
+  on x._home {
     qio_channel_retain(x._channel_internal);
   }
 }
 
-proc channel.init=(x: channel) {
+proc _channel.init=(x: _channel) {
```

Comment:
I think this needs a `pragma "no doc"`, otherwise it turns up in the documentation (and exposes the `_channel` type as a result)